.github/workflows/tests.yml

```diff
@@ -64,6 +64,17 @@ jobs:
             python-version: "3.8"
             nocython: 1
 
+          # Python 3.10 and numpy 1.22
+          # Use conda-forge to provide numpy 1.22
+          # Ignore ImportWarning because pyximport registered an importer
+          # PyxImporter that does not have a find_spec method and this raises
+          # a warning on Python 3.10
+          - case-name: Python 3.10
+            os: ubuntu-latest
+            python-version: "3.10"
+            condaforge: 1
```

Comment:
The conda-forge numpy doesn't seem to find / use mkl even if the mkl blas library is installed (or at least qutip.settings.has_mkl remains False and numpy's config has blas set to Generic). I'm don't know if this is expected or not. Perhaps we should explicitly specify `nomkl: 1` here to make that clear.